File: tauri-template/tests/url-validator.test.ts
import { describe, expect, it } from 'vitest';
import { displayHost, validateUrl } from '../src/lib/url-validator';

describe('validateUrl', () => {
  it('accepts https URL', () => {
    const result = validateUrl('https://app.example.com');
    expect(result.isValid).toBe(true);
    expect(result.normalizedUrl).toBe('https://app.example.com/');
  });

  it('adds https scheme when missing', () => {
    const result = validateUrl('app.example.com');
    expect(result.isValid).toBe(true);
    expect(result.normalizedUrl).toBe('https://app.example.com/');
  });

  it('rejects empty input', () => {
    const result = validateUrl('   ');
    expect(result.isValid).toBe(false);
  });

  it('rejects unsupported scheme', () => {
    const result = validateUrl('ftp://app.example.com');
    expect(result.isValid).toBe(false);
  });
});

describe('displayHost', () => {
  it('returns hostname', () => {
    expect(displayHost('https://app.example.com/path')).toBe('app.example.com');
  });
});

File: flutter-template/test/url_validator_test.dart
import 'package:flutter_test/flutter_test.dart';
import 'package:webapp_template/core/utils/url_validator.dart';

void main() {
  group('UrlValidator', () {
    test('accepts https URL', () {
      final result = UrlValidator.validate('https://app.example.com');
      expect(result.isValid, isTrue);
      expect(result.normalizedUrl, 'https://app.example.com/');
    });

    test('adds https scheme when missing', () {
      final result = UrlValidator.validate('app.example.com');
      expect(result.isValid, isTrue);
      expect(result.normalizedUrl, 'https://app.example.com/');
    });

    test('rejects empty input', () {
      final result = UrlValidator.validate('   ');
      expect(result.isValid, isFalse);
      expect(result.error, isNotNull);
    });

    test('rejects unsupported scheme', () {
      final result = UrlValidator.validate('ftp://app.example.com');
      expect(result.isValid, isFalse);
    });

    test('displayHost returns hostname', () {
      expect(
        UrlValidator.displayHost('https://app.example.com/path'),
        'app.example.com',
      );
    });
  });
}
